prow/tide/history/history.go

```diff
@@ -62,30 +45,26 @@ type History struct {
 	sync.Mutex
 	logSizeLimit int
 
-	gcsObj *storage.ObjectHandle
+	opener io.Opener
+	path   string
 }
 
-func readHistory(maxRecordsPerKey int, obj storageObject) (map[string]*recordLog, error) {
-	reader, err := obj.NewReader()
-	if err == storage.ErrObjectNotExist {
-		// No history exists yet. This is not an error.
+func readHistory(maxRecordsPerKey int, opener io.Opener, path string) (map[string]*recordLog, error) {
+	reader, err := opener.Reader(context.Background(), path)
+	if io.IsNotExist(err) { // No history exists yet. This is not an error.
 		return map[string]*recordLog{}, nil
 	}
 	if err != nil {
-		return nil, fmt.Errorf("error opening GCS object: %v", err)
+		return nil, fmt.Errorf("open: %v", err)
 	}
-	defer func() {
-		if err := reader.Close(); err != nil {
-			logrus.WithError(err).Error("Error closing GCS object reader.")
-		}
-	}()
+	defer io.LogClose(reader)
 	raw, err := ioutil.ReadAll(reader)
 	if err != nil {
-		return nil, fmt.Errorf("error reading GCS object: %v", err)
+		return nil, fmt.Errorf("read: %v", err)
 	}
 	var recordsByPool map[string][]*Record
 	if err := json.Unmarshal(raw, &recordsByPool); err != nil {
-		return nil, fmt.Errorf("error unmarshaling GCS object: %v", err)
+		return nil, fmt.Errorf("unmarshal: %v", err)
```

Comment:
Do we get more context somewhere? This is a bit terse

Reply:
Yes, this will become:
* error initializing history client from `/some/path`: unmarshal: something interesting from json"

see Note below